cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -298,3 +307,19 @@ func getBucketListSize(ctx context.Context, ledgerReader db.LedgerReader, latest
 	}
 	return uint64(closeMeta.V1.TotalByteSizeOfBucketList), nil
 }
+
+var cachedCoreProtocolVersion atomic.Int32
+
+// getProtocolVersion obtains the memoized protocol version from a core client
+func getProtocolVersion(ctx context.Context, client interfaces.CoreClient) (int, error) {
+	version := cachedCoreProtocolVersion.Load()
+	if version != 0 {
+		return int(version), nil
+	}
+	info, err := client.Info(ctx)
+	if err != nil {
+		return 0, err
+	}
+	cachedCoreProtocolVersion.CompareAndSwap(0, int32(info.Info.ProtocolVersion))
+	return info.Info.ProtocolVersion, nil
+}
```

Comment:
Is requesting and caching forever too much? I suppose Core could be restarted due to an error, but I doubt it would magically change the protocol version?

Reply:
I think something like that could potentially happen in Horizon, so I am not sure. @tamirms thoughts?

Reply:
Where exactly does the protocol version come from in the `CoreClient`? During the protocol upgrade the version will change, without any restarts.

Reply:
@2opremio I see that you already get the ledger close meta and use it to configure the `BucketListSize`. The the ledger close meta also contains the protocol version (it is located in the ledger header `closeMeta.LedgerHeaderHistoryEntry().Header.LedgerVersion`). I think it would be better to obtain the protocol version from the ledger close meta instead of querying core's http endpoint

Reply:
Ah ... really good point. Thanks!